File: Cpp/1380_Lucky_Numbers_in_a_Matrix.cpp
class Solution {
public:
    vector<int> luckyNumbers (vector<vector<int>>& matrix) {
        vector<int> result;
        int m = matrix.size();
        int n = matrix[0].size();
        
        for(int i=0; i<m; ++i){
            int minimum = INT_MAX;
            int index = 0;
            for(int j=0; j<n; ++j){
                if(minimum > matrix[i][j]){
                    minimum = matrix[i][j];
                    index = j;
                }
            }
            int maximum = minimum;
            for(int j=0; j<m; ++j){
                maximum = max(maximum, matrix[j][index]);
            }
            
            if(minimum == maximum){
                result.push_back(minimum); 
            }
        }
        
        return result;
    }
};

File: Kotlin/1380_Lucky_Numbers_in_a_Matrix.kt
class Solution {
    fun luckyNumbers (matrix: Array<IntArray>): List<Int> {
        val result = mutableListOf<Int>()
        val m = matrix.size
        val n = matrix[0].size
        
        for(i in 0 until m) {
            var minimum = Int.MAX_VALUE
            var index = 0
            for(j in 0 until n) {
                if(minimum > matrix[i][j]){
                    minimum = matrix[i][j]
                    index = j
                }
            }
            var maximum = minimum
            for(j in 0 until m) {
                maximum = maxOf(maximum, matrix[j][index])
            }
            
            if(minimum == maximum) {
                result.add(minimum)
            }
        }
        
        return result
    }
}